wenet/bin/export_onnx_cpu.py

```diff
@@ -0,0 +1,327 @@
+
+    model = init_asr_model(configs)
+    load_checkpoint(model, args.checkpoint)
+    model.eval()
+    print(model)
+
+    batch = 1
+    chunk_size = args.chunk_size
+    left_chunks = args.num_decoding_left_chunks
+    beam = args.beam
+    reverse_weight = args.reverse_weight
+    output_size = configs['encoder_conf']['output_size']
+    num_blocks = configs['encoder_conf']['num_blocks']
+    cnn_module_kernel = configs['encoder_conf']['cnn_module_kernel']
+    head = configs['encoder_conf']['attention_heads']
+    feature_size = configs['input_dim']
+    vocab_size = configs['output_dim']
+    # NOTE(xcsong): if chunk_size == -1, hardcode to 67
+    decoding_window = (chunk_size - 1) * model.encoder.embed.subsampling_rate \
+        + model.encoder.embed.right_context + 1 if chunk_size > 0 else 67
+
+    # ============================= encoder =================================
+    print("\033[32mStage-1: export encoder\033[0m")
+    encoder = model.encoder
+    encoder.forward = encoder.forward_chunk
+    encoder_outpath = os.path.join(output_dir, 'encoder.onnx')
+
+    print("\t\033[32mStage-1.1: prepare inputs for encoder\033[0m")
+    chunk = torch.randn((batch, decoding_window, feature_size))
+    offset = 0
+    # NOTE(xcsong): if left_chunks > 0, we feed real cache & real mask
+    #   to the model, otherwise fake cache & fake mask. This is to ensure
+    #   that all if-else branches in `forward_chunk` will always take the
+    #   same path with a given setup(i.e., 16/-1, 16/4, 16/0 or -1/-1)
+    if left_chunks > 0:  # 16/4
+        required_cache_size = chunk_size * left_chunks
+        offset = required_cache_size
+        # Real cache
+        att_cache = torch.zeros(
+            (num_blocks, head, required_cache_size, output_size // head * 2))
+        # Real mask
+        att_mask = torch.ones(
+            (batch, 1, required_cache_size + chunk_size), dtype=torch.bool)
+        att_mask[:, :, :required_cache_size] = 0
+    elif left_chunks <= 0:  # 16/-1, -1/-1, 16/0
+        required_cache_size = -1 if left_chunks < 0 else 0
+        # Fake cache
+        att_cache = torch.zeros(
+            (num_blocks, head, 0, output_size // head * 2))
+        # Fake mask
+        att_mask = torch.ones((0, 0, 0), dtype=torch.bool)
+    cnn_cache = torch.zeros(
+        (num_blocks, batch, output_size, cnn_module_kernel - 1))
+    inputs = (chunk, offset, required_cache_size,
+              att_cache, cnn_cache, att_mask)
+    print("\t\t\033[32mchunk.size(): {}\033[0m\n".format(chunk.size()),
+          "\t\t\033[32moffset: {}\033[0m\n".format(offset),
+          "\t\t\033[32mrequired_cache: {}\033[0m\n".format(required_cache_size),
+          "\t\t\033[32matt_cache.size(): {}\033[0m\n".format(att_cache.size()),
+          "\t\t\033[32mcnn_cache.size(): {}\033[0m\n".format(cnn_cache.size()),
+          "\t\t\033[32matt_mask.size(): {}\033[0m\n".format(att_mask.size()))
+
+    print("\t\033[32mStage-1.2: torch.onnx.export\033[0m")
```

Comment:
replace print("\t\033[32m{}\033[0m") with a function

Reply:
Can we split the whole processing into functions, seems it is too long now.
I think we can move the verfication code into a new function.

Reply:
done